OneDrive - w5fls/Desktop/potpie-main/potpie-main/app/modules/code_provider/provider_factory.py

```
@@ -0,0 +1,459 @@
+            private_key = f"-----BEGIN RSA PRIVATE KEY-----\n{private_key}\n-----END RSA PRIVATE KEY-----\n"
+
+        auth = AppAuth(app_id=app_id, private_key=private_key)
+        jwt = auth.create_jwt()
+
+        url = f"https://api.github.com/repos/{repo_name}/installation"
+        headers = {
+            "Accept": "application/vnd.github+json",
+            "Authorization": f"Bearer {jwt}",
+            "X-GitHub-Api-Version": "2022-11-28",
+        }
+        response = requests.get(url, headers=headers)
```

Ruff (0.14.8)

Probable use of `requests` call without timeout

(S113)